projects/firmware_homework/src/main.c

```diff
@@ -0,0 +1,74 @@
+// MidSun libraries
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "wait.h"
+
+// Interrupt callback function
+static void prv_gpio_callback(const GpioAddress *address, void *context) {
+  // Context should be the address of A6 as we want to read from this pin
+  GpioAddress *adc_channel = context;
+  uint16_t adc_data = 0;
+  // Read the values from the ADC channel and output the reading
+  adc_read_converted_pin(*adc_channel, &adc_data);
```

Comment:
Though this is okay. What would be even better is if we add a check to see what the `StatusCode` is which is returned from this function to make sure that `adc_data` is valid.

Reply:
Is there a way to confirm that adc_data is valid? I'm sure there is a module but I didn't see mentions of it in the tutorial.

Reply:
Lots of our functions return a `StatusCode`, which is a return code that tells you whether everything went alright in the function. It's our alternative to exceptions as found in other languages. Generally if a function returns `STATUS_CODE_OK`, everything went alright, and if it returns another code then you should do any applicable cleanup and pass the status code up the stack by returning it. See status.h (in libraries/libcore) for the list of status codes, and check out the `status_ok_or_return` macro.